moabb/datasets/Hinss2021.py

```diff
@@ -0,0 +1,175 @@
+
+    .. admonition:: Dataset summary
+
+
+        ==============  =======  =======  ==========  =================  ============  ===============  ===========
+        Name              #Subj    #Chan    #Classes    #Blocks/class     Trials len    Sampling rate    #Sessions
+        =============== =======  =======  ==========  =================  ============  ===============  ===========
+        Hinss2021            15       62           4                 1      2s            250Hz                 2
+        =============== =======  =======  ==========  =================  ============  ===============  ===========
+
+    We describe the experimental procedures for a dataset that is publicly available
+    at https://zenodo.org/records/5055046.
+    This dataset contains electroencephalographic recordings of 15 subjects (6 female, with an
+    average age of 25 years). A total of 62 active Ag–AgCl
+    electrodes were available in the dataset.
+
+    The participants engaged in 3 (2 available here) distinct experimental sessions, each of which
+    was separated by 1 week.
+
+    At the beginning of each
+    session, the resting state of the participant (measured as
+    1 minute with eyes open) was recorded.
+
+    Subsequently, participants undertook 3 tasks of varying difficulty levels
+    (i.e., easy, medium, and difficult). The task assignments
+    were randomized.
+
+    Notes
+    -----
+
+    .. versionadded:: 1.0.1
+
+    References
+    ----------
+
+    .. [1] M. Hinss, B. Somon, F. Dehais & R. N. Roy (2021)
+            Open EEG Datasets for Passive Brain-Computer
+            Interface Applications: Lacks and Perspectives.
+            IEEE Neural Engineering Conference.
+    """
+
+    def __init__(self):
+        super().__init__(
+            subjects=list(range(1, 16)),  # 15 participants
+            sessions_per_subject=2,  # 2 sessions per subject
+            events=EVENTS,
+            code="Hinss2021",
+            interval=[0, 2],  # Epochs are 2-second long
+            paradigm="rstate",
+        )
+
+    def _get_stim_channel(self, rs_epochs, easy_epochs, med_epochs, n_epochs, n_samples):
+        n_epochs_rs = rs_epochs.get_data().shape[0]
+        n_epochs_easy = easy_epochs.get_data().shape[0]
+        n_epochs_med = med_epochs.get_data().shape[0]
+        stim = np.zeros((1, n_epochs * n_samples))
+        for i in range(n_epochs):
+            stim[0, n_samples * i + 1] = (
+                EVENTS["rs"]
+                if i < n_epochs_rs
+                else (
+                    EVENTS["easy"]
+                    if i < n_epochs_rs + n_epochs_easy
+                    else (
+                        EVENTS["medium"]
+                        if i < n_epochs_rs + n_epochs_easy + n_epochs_med
+                        else EVENTS["diff"]
+                    )
+                )
+            )
+        return stim
+
+    def _get_epochs(self, session_path, subject, session, event_file):
```

Comment:
Is this epoch file pre-processed by the authors?

Reply:
I don't think it's a good idea to read epoched data. But I don't know, copying Sylvain to help @sylvchev.

Reply:
Yes, they are preprocessed. I did not see the raw files in the online dataset.

Reply:
I talked with @sylvchev, and it is okay to integrated the epoched format

Reply:
Ok, nice. Do you need additional changes before merging?